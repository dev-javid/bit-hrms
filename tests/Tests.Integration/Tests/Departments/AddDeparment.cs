namespace Tests.Integration.Tests.Departments
{
    public class AddDeparment(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/departments";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            var command = new
            {
                Name = Faker.Random.String(3, 50)
            };
            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PostAsync(Route, content);
            }

            var result = await response.Content.ReadAsStringAsync();
            response.Dispose();
            Approvals.VerifyJson(result.Ignore("$.departmentId", "$.name"));
        }
    }
}
