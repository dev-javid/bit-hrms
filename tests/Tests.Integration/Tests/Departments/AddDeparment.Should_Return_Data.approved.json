{
    "departmentId": "__IGNORED_VALUE__",
    "name": "__IGNORED_VALUE__",
    "jobTitles": []
}